codegen/src/main/java/com/paypal/selion/plugins/CodeGeneratorMojo.java

```diff
@@ -124,6 +125,11 @@ private String pathToFolder(File file) {
      */
     public void execute() throws MojoExecutionException {
 
+        // iff basePackage is null, empty, or just whitespace reset it to the default value
+        basePackage = StringUtils.defaultString(basePackage, "com.paypal.selion.testcomponents");
```

Comment:
@fakrudeen78 How about 
```java
if (basePackage.isEmpty() && folder.isEmpty()) {
  folder = "default";
}
``` 
I believe this approach will meet my ask w.r.t convention, allows for the user to define a detailed folder structure under `baseFolder` and leave `basePackage` empty, accommodates any files directly in the `baseFolder`, and avoids any exception.